From: Brick Linux <brick@localhost>
Subject: [PATCH] httpd: restore the index page name when only CGI is enabled

handle_incoming_and_exit() cuts ".../index.html" back to ".../" for
check_user_passwd() when BASIC_AUTH *or* CGI is enabled, but only put the
name back when BASIC_AUTH is enabled. With CGI on and BASIC_AUTH off, every
directory URL ("/") was served as an empty path and returned 404.

--- a/networking/httpd.c
+++ b/networking/httpd.c
@@ -2614,7 +2614,7 @@ static void handle_incoming_and_exit(const len_and_sockaddr *fromAddr)
 	/* !CGI: it can be only GET or HEAD */
 #endif
 
-#if ENABLE_FEATURE_HTTPD_BASIC_AUTH
+#if ENABLE_FEATURE_HTTPD_BASIC_AUTH || ENABLE_FEATURE_HTTPD_CGI
 	/* Restore truncated .../index.html */
 	if (urlp[-1] == '/')
 		urlp[0] = index_page[0];
